core/chains/evm/headtracker/head_tracker.go

```diff
@@ -39,25 +41,67 @@ var (
 // HeadsBufferSize - The buffer is used when heads sampling is disabled, to ensure the callback is run for every head
 const HeadsBufferSize = 10
 
-type headTracker struct {
+type headTracker[
+	HTH htrktypes.Head[BLOCK_HASH, ID],
+	S commontypes.Subscription,
+	ID txmgrtypes.ID,
+	BLOCK_HASH commontypes.Hashable,
+] struct {
 	log             logger.Logger
-	headBroadcaster httypes.HeadBroadcaster
-	headSaver       httypes.HeadSaver
+	headBroadcaster commontypes.HeadBroadcaster[HTH, BLOCK_HASH]
+	headSaver       commontypes.HeadSaver[HTH, BLOCK_HASH]
 	mailMon         *utils.MailboxMonitor
-	ethClient       evmclient.Client
-	chainID         big.Int
+	ethClient       htrktypes.Client[HTH, S, ID, BLOCK_HASH]
+	chainID         txmgrtypes.ID
```

Comment:
this should be ID?

Reply:
my mistake for leaving that in, thanks for pointing it out. will keep that in mind in the future.

Reply:
my mistake for leaving that in, thanks for pointing it out. will keep that in mind in the future.